python/plugins/processing/algs/gdal/ClipRasterByMask.py

```diff
@@ -281,6 +283,72 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         out = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
         self.setOutputValue(self.OUTPUT, out)
 
+        # If it's a WMS layer and scale/DPI were given, craft a XML description file
+        if inLayer.providerType() == "wms" and isinstance(
```

Comment:
Can you move all this logic to gdal_connection_details_from_layer ?

Reply:
Hm.... couldn't you just make gdal_connection_details_from_layer accept an optional bbox parameter then? And then the algorithm specific logic for calculating that bbox can live in the algorithm, but the generic bits about QgsProcessingRasterLayerDefinition values would live in the common method